video/decode/videotoolbox.c

```diff
@@ -100,10 +123,93 @@ static int init_decoder(struct lavc_ctx *ctx, int w, int h)
     return 0;
 }
 
+static int init_decoder_copy(struct lavc_ctx *ctx, int w, int h)
+{
+    av_videotoolbox_default_free(ctx->avctx);
+
+    int err = av_videotoolbox_default_init(ctx->avctx);
+    if (err < 0) {
+        print_videotoolbox_error(ctx->log, MSGL_ERR, "failed to init videotoolbox decoder", err);
+        return -1;
+    }
+
+    return 0;
+}
+
```

Comment:
Can you try to also reduce code duplication here? (I realize it's not much code, and the pixel format selection seems like a bit of a pain.)